File: leetcode/containerWithMostWater.cpp
//============================================================================
// Name        :
// Author      : hussam
// Version     :
// Copyright   :
// Description :https://leetcode.com/problems/container-with-most-water/description/
//============================================================================

class Solution {
public:
   int maxArea(vector<int>& height) {
    int res=0,left=0,right=height.size()-1;
    
    while(left<right)
    {
    		res=max(res,(right-left)*min(height[left],height[right]));
    		
    		if(height[left]<height[right])
    		left++;
    		else
    		right--;
    		
    	
    }
	
	return res;
   }

};

File: leetcode/two pointers/meeting array/container_with_most_water.py
# description : https://leetcode.com/problems/container-with-most-water/description/ # NOQA
# author : Hussam
# solution : two pointers
# o(n) time, o(1) space


class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left, right = 0, len(height)-1
        water_amount = 0
        while left < right:
            water_amount = max(water_amount,
                               min(height[left], height[right]) *
                               (right-left)
                               )
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return water_amount
